File: PYTHON/fold-array.py
# n=int(input())
# arr=[]
# for i in range(n):
#     arr.append(int(input()))
# fold=int(input())
# if(fold<=0):
#     print(fold)
#     for i in range(n):
#         print(arr[i])
# else:
#     orgArr=[]
#     for k in range(fold):
#         arr1=[]
#         for i,j in range(0,len(arr)-1,j+1):
#             if(len(arr)==1):
#                 arr1.append(arr[i])
#             elif(i<j):
#                 arr1.append(arr[i]+arr[j])
#             elif(i==j):
#                 arr1.append(arr[i])
#         arr=arr1
#         n=len(arr1)
# print(len(arr))
# for i in range(len(arr)):
#     print(arr[i])

n =int(input())
arr=[]
for i in range(n):
    arr.append(int(input()))
fold =int(input())
for i in range(fold):
    arr2=[]
    i=0
    j=len(arr)-1
    while(i<j):
        arr[i] +=arr[j]
        arr.pop()
        j =j-1
        i =i+1
print(len(arr))
for i in arr:
    print(i)

File: ACA/fold-array.js
let fs = require("fs");
let data = fs.readFileSync("C:/Users/ASUS/OneDrive/Desktop/10x/input.txt", 'utf-8');
let idx = 0;
data = data.split('\n');

function readLine() {
    idx++;
    return data[idx - 1].trim();
}
let n = parseInt(readLine());
let arr = [];
for (i = 0; i < n; i++) {
    arr.push(parseInt(readLine()));
}
let fold = parseInt(readLine());
if (fold <= 0) {
    console.log(fold);
    for (let i = 0; i < n; i++) {
        console.log(arr[i]);
    }
} else {
    let orgArr = [];
    for (let k = 0; k < fold; k++) {
        let arr1 = [];
        for (let i = 0, j = arr.length - 1; i <= j; i++, j--) {
            if (arr.length == 1) {
                arr1.push(arr[i]);
            } else if (i < j) {
                arr1.push(arr[i] + arr[j]);
            } else if (i = j) {
                arr1.push(arr[i]);
            }
        }
        arr = arr1;
        n = arr1.length;
        // orgArr = arr1;
    }
    //     console.log(orgArr.length);
    //     for (i = 0; i < orgArr.length; i++) {
    //         console.log(orgArr[i]);
    //     }   console.log(orgArr.length);
    console.log(arr.length)
    for (i = 0; i < arr.length; i++) {
        console.log(arr[i]);
    }

}